text/batch-l2.md

```diff
@@ -0,0 +1,1540 @@
+(25% reserved + 25% unreserved).
+
+### Configuring Job Retry Policies
+
+Certain workflows may result in Jobs failing due to intermittent issues.
+Jobs can specify retry policies to respond to different failures with different actions.
+There are three different ways information about the way a Job exited can be conveyed;
+
+* `exitCode`: the exit code returned from the process executed by the container. Will only match non-zero exit codes.
+* `reason`: any middleware errors, like your Docker registry being down.
+* `statusReason`: infrastructure errors, most commonly your spot instance being reclaimed. 
+
+You can specify a glob string to match each of these and react to different failures accordingly.
+Up to five different retry strategies can be configured for each Job,
+and each strategy can match against some or all of `exitCode`, `reason`, and `statusReason`.
+You can optionally configure the number of times a job will be retried,
+but you cannot configure different retry counts for different strategies; they all share the same count.
+If multiple conditions are specified in a given retry strategy,
+they must all match for the action to be taken; the conditions are ANDed together, not ORed.
+This example configures four retry strategies:
+
+```ts
+const jobDefn = new batch.EcsJobDefinition(this, 'JobDefn', {
+   containerDefinition: new batch.ContainerDefinition(this, 'containerDefn', {
+    image: ecs.ContainerImage.fromRegistry('public.ecr.aws/amazonlinux/amazonlinux:latest'),
+    memoryLimitMiB: 2048,
+    compatibility: Compatibility.EC2, // specify FARGATE for Fargate workflows
+  }),
+  attempts: 5,
+  retryStrategies: [{
+    action: Action.EXIT,
+    onExitCode: '40*', // matches 40, 400, 404, 40230498, etc
+  }],
+});
+
+jobDefn.addRetryStrategy({
+  action: Action.RETRY,
+  onStatusReason: StatusReason.SPOT_INSTANCE_RECLAIMED,
+});
+
+jobDefn.addRetryStrategy({
+   action: Action.EXIT,
+   onReason: Reason.CANNOT_PULL_CONTAINER,
+});
+
+jobDefn.addRetryStrategy({
+  action: Action.RETRY,
+  onStatusReason: StatusReason.custom('Some other reason*'),
+});
+```
+
+### Running single-container ECS workflows
+
+Batch supports ECS workflows natively. This examples creates a `JobDefinition` that runs a single container with ECS:
+
+```ts
+const jobDefn = new batch.EcsJobDefinition(this, 'JobDefn', {
+  containerDefinition: new batch.ContainerDefinition(this, 'containerDefn', {
+    image: ecs.ContainerImage.fromRegistry('public.ecr.aws/amazonlinux/amazonlinux:latest'),
+    memoryLimitMiB: 2048,
+    compatibility: Compatibility.EC2, // specify FARGATE for Fargate workflows
+  }),
+});
+```
+
+For workflows that need persistent storage, batch supports mounting `Volume`s to the container.
+You can both provision the volume and mount it to the container in a single operation:
+
+```ts
+jobDefn.addVolume({
+  name: 'myVolume',
+  efsVolumeConfiguration: {
+    fileSystem: myFileSystem
+  },
```

Comment:
```suggestion
  volume: Volume.efs(myFileSystem),
```

Reply:
I'll address this tomorrow, I think there's some room to improve on this API (I basically lifted it from aws-ecs)